### PR TITLE
Improved wake behaviour

### LINKED ISSUE #82: Display Wakeup only for active display element changes
**Is your feature request related to a problem? Please describe.**
Psudo problem: I used discord frequently and have ducking enabled. This changes the active volume of all my audio sessions when someone is speaking. This in turn triggers the display to wake up, even if the display is currently on the Master volume which does not change. So the result is the device is actively waking up very frequently and not much I can do about it. This is also not very great for the OLED screens as they are extremely sensitive to burn in.  

**Describe the solution you'd like**
The display should not wake up unless the command received is for the current active item on the display. This allows you to switch to the Master volume, and let it go to sleep as it's volume doesn't change during these ducking sessions.

**Describe alternatives you've considered**

**Additional context**
Keyboard OLED displays use timeouts and display scrolling to handle the same burn in issue:
https://beta.docs.qmk.fm/using-qmk/hardware-features/displays/feature_oled_driver

Thoughts?

### PR DESCRIPTION
**Pull request by SchemingWeasels**
## Issues
 - Resolves #82 

## Description
Reduced screen wakes to only relevant events

## Types of changes
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] Requested changes are in a branch
- [X] Compiled and tested requested changes on target hardware (PC, device)
- [X] Updated the documentation, if necessary
- [ ] Updated the LICENSES file, if necessary
- [X] Reviewed the [guidelines for contributing](../CONTRIBUTING.md) to this repository

## Further comments

Not sure if changing the way the command decode is ideal. I did it this way because I need access to the incoming command.
The other option would be to put in the wake-checking code into ProcessPackage, which seems like a worse idea to me. Thoughts?

### CODE REVIEW

File: Embedded/MaxMix/MaxMix.ino [hunk 1]
```diff
@@ -513,3 +523,23 @@ int8_t FindItem(uint32_t id)
 
   return -1;
 }
+
+//---------------------------------------------------------
+// Checks if command package target is the active application.
+// Returns true if ids match.
+//---------------------------------------------------------
+bool IsActive()
```
Comment (t3knomanzer) [inline Embedded/MaxMix/MaxMix.ino]: I would probably rename this to something a bit less generic such as IsCommandItemActive or something like that.

File: Embedded/MaxMix/MaxMix.ino [hunk 2]
```diff
@@ -128,11 +128,23 @@ void loop()
...
+      if(command == MSG_COMMAND_UPDATE_VOLUME && IsActive())
+      {
+        UpdateActivityTime();
+        isDirty = true;
+      }
+      else if(command == MSG_COMMAND_ADD && settings.displayNewSession)
+      {
+        UpdateActivityTime();
+        isDirty = true;
+      }
```
Comment (t3knomanzer) [inline Embedded/MaxMix/MaxMix.ino]: I agree with your observation. I would probably rename UpdateActivityTime to UpdateActivity and move all this logic inside.

**Review by t3knomanzer — COMMENTED** [review]
@XScorpion2 This was your request, what do you think?

**Comment by XScorpion2** [comment]
So I haven't spent a whole lot of time looking at the firmware code layout, but over all the approach makes sense so I'm good with it.

**Comment by t3knomanzer** [comment]
Ok. I'm going to wait to merge this until after we release 0.10.5. I'm currently testing it and fixing some bugs introduced before releasing it.

**Comment by SchemingWeasels** [comment]
I have completely restructured the way the checks are performed. ProcessPackage returns true if an update is required, just like the other process functions. Having said what I said in the original post, I think this is a cleaner approach, and it kinda makes sense to make ProcessPackage flag if a screen update is required. 

What do you think?

I also renamed the functions to be more descriptive as suggested and did some documentation for the functions I changed.

**Comment by SchemingWeasels** [comment]
> and fixing some bugs introduced before releasing it.

Is it on the firmware side?

**Comment by t3knomanzer** [comment]
No, it’s on the application side. We introduced process grouping which involved a pretty big rewrite of the audio service and that’s introduced some issues. 
It’s almost done now.

**Comment by t3knomanzer** [comment]
Hey, can you rebase on master so I can test it with the latest changes?

**Review by t3knomanzer — APPROVED** [review]

**Pull request merged**